src/module.ts

```diff
@@ -25,13 +28,26 @@ export default defineNuxtModule<ModuleOptions>({
     siteKey: nuxt.options.dev ? '1x00000000000000000000AA' : undefined,
     addValidateEndpoint: false,
   }),
-  async setup (options, nuxt) {
+  async setup(options, nuxt) {
     const siteKey = options.siteKey || nuxt.options.runtimeConfig.public?.turnstile?.siteKey
     if (!siteKey) {
       console.warn('`@nuxtjs/turnstile` is disabled as no site key was provided.')
       return
     }
 
+    const secretKeyPath =
+      options.secretKeyPath || nuxt.options.runtimeConfig.turnstile?.secretKeyPath
+    let secretKeyFromPath
+    if (secretKeyPath) {
+      secretKeyFromPath = fs.existsSync(secretKeyPath)
+        ? fs.readFileSync(secretKeyPath, 'utf-8')
+        : undefined
+    }
+    if (!secretKeyFromPath) {
+      console.warn('The secret key from the path could not be found.')
```

Comment:
Wouldn't this be shown every time someone uses a `secretKey` only, without reading from file?

Reply:
As far as I can see the setup function is called once when the module is loaded. But maybe I'm wrong.

Reply:
Yes, but should it show a warning if a user never intended to make use of secret files?

Reply:
I See. I'll change that so the warning will only be shown when the directory of the file for the secret key is passed as an input

Reply:
This thread can be resolved :+1: